crypto/evp/evp_enc.c

```diff
@@ -605,6 +862,22 @@ int EVP_CIPHER_CTX_set_padding(EVP_CIPHER_CTX *ctx, int pad)
         ctx->flags &= ~EVP_CIPH_NO_PADDING;
     else
         ctx->flags |= EVP_CIPH_NO_PADDING;
+
+    if (ctx->cipher != NULL && ctx->cipher->prov != NULL) {
+        const OSSL_PARAM params[] = {
+            OSSL_PARAM_int(OSSL_CIPHER_PARAM_PADDING, &pad),
+            OSSL_PARAM_END
+        };
+
+        if (ctx->cipher->ctx_set_params == NULL) {
+            EVPerr(EVP_F_EVP_CIPHER_CTX_SET_PADDING, EVP_R_CTRL_NOT_IMPLEMENTED);
+            return 0;
+        }
+
+        if (!ctx->cipher->ctx_set_params(ctx->provctx, params))
```

Comment:
Should this just return whtever the set_params function returned?

Reply:
No - I don't think so. I prefer to be explicit that the return from this function is either 0 or 1 - not whatever truth value the provider happens to use.